oommfc/drivers/hysteresisdriver.py

```diff
@@ -58,14 +70,44 @@ class HysteresisDriver(Driver):
         "report_wall_time",
     ]
 
-    def _checkargs(self, **kwargs):
-        Hmin, Hmax, n = kwargs["Hmin"], kwargs["Hmax"], kwargs["n"]
-        for i in [Hmin, Hmax]:
+    def _checkargs(self, kwargs):
+        # check the default arguments for a symmetric hysteresis loop
+        if any(item in kwargs for item in ["Hmin", "Hmax", "n"]) and "Hsteps" in kwargs:
+            msg = "Cannot define both (Hmin, Hmax, n) and Hsteps."
+            raise ValueError(msg)
+
+        if all(item in kwargs for item in ["Hmin", "Hmax", "n"]):
```

Comment:
This check implies that `(Hmin, Hmax, n)` has precedence over the new `Hstep` and we do not warn/fail if both are defined. Are we okay with this @samjrholt and @swapneelap ?

Reply:
I would have a small preference for only being able to define one at a time.

Reply:
I'm confused, this first check of this arg-check function implies that if any of (Hmin, Hmax and n) are mentioned together with Hsteps, an exception will be raised, It does exactly what @samjrholt would want to?

Reply:
You are right, I missed that, sorry. Could you add it to this if-elif-else block (as the first if) so that it is more obvious that it is part of the block?

Reply:
Yeah sure 👍🏼